File: src/main/java/com/magichamster/grocerysamurai/model/StoreItem.java
package com.magichamster.grocerysamurai.model;

import javax.persistence.*;

/**
 * Allow for items on different aisles Created by developer on 4/14/17.
 */
@Entity
@Table(name = "store_item")
public class StoreItem extends Identity {
	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name = "store_id", nullable = false)
	private Store store;

	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name = "aisle_id", nullable = false)
	private Aisle aisle;

	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name = "item_id", nullable = false)
	private Item item;

	public StoreItem() {
	}

	public Store getStore() {
		return store;
	}

	public void setStore(Store store) {
		this.store = store;
		store.getStoreItems().add(this);
	}

	public Aisle getAisle() {
		return aisle;
	}

	public void setAisle(Aisle aisle) {
		this.aisle = aisle;
		aisle.getStoreItems().add(this);
	}

	public Item getItem() {
		return item;
	}

	public void setItem(Item item) {
		this.item = item;
		item.getStoreItems().add(this);
	}
}

File: MagicHamster.GrocerySamurai/MagicHamster.GrocerySamurai.Model/Entities/StoreItem.cs
namespace MagicHamster.GrocerySamurai.Model.Entities
{
    using System.ComponentModel.DataAnnotations.Schema;
    using Common;
    using JetBrains.Annotations;

    [Table("store_item")]
    public class StoreItem : Entity
    {
        [Column("store_id")]
        [UsedImplicitly]
        public int StoreId { get; set; }

        [Column("aisle_id")]
        [UsedImplicitly]
        public int AisleId { get; set; }

        [Column("item_id")]
        [UsedImplicitly]
        public int ItemId { get; set; }

        [ForeignKey("StoreId")]
        [UsedImplicitly]
        public Store Store { get; set; }

        [ForeignKey("AisleId")]
        [UsedImplicitly]
        public Aisle Aisle { get; set; }

        [ForeignKey("ItemId")]
        [UsedImplicitly]
        public Item Item { get; set; }
    }
}
